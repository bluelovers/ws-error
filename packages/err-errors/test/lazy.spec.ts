import getSubErrors from '../index';
import { AggregateError as BluebirdAggregateError } from 'bluebird';
import { basename } from 'path';

const arr = [new Error('1'), new Error('3')] as const;

describe(basename(__filename), () =>
{


	test(`AggregateError`, () =>
	{
		let e = new AggregateError(arr);

		let actual = _check(e);

		expect(actual).toStrictEqual(e.errors);

	});

	test(`Bluebird.AggregateError`, () =>
	{
		let e = new BluebirdAggregateError();

		e.push(...arr);

		let actual = _check(e);

		expect(actual).toStrictEqual(e.slice());

	});

})

function _check(e: Error)
{
	let actual = getSubErrors(e);

	expect(actual).toStrictEqual(arr);
	expect(actual).toMatchSnapshot();

	return actual
}
