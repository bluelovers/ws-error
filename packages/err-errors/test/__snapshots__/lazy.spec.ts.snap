// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lazy.spec.ts AggregateErrorExtra 1`] = `
Array [
  [Error: 1],
  [Error: 3],
]
`;

exports[`lazy.spec.ts AggregateErrorExtra 2`] = `
Array [
  [Error: 1],
  [Error: 3],
]
`;

exports[`lazy.spec.ts Bluebird.AggregateError 1`] = `
Array [
  [Error: 1],
  [Error: 3],
]
`;

exports[`lazy.spec.ts es.AggregateError 1`] = `
Array [
  [Error: 1],
  [Error: 3],
]
`;
